### PR TITLE
Fix #18971: New Game does not prompt for save before quitting

### LINKED ISSUE #18971: Game allows starting new game without prompting for save
### Operating System

Windows 10, 64-bit

### OpenRCT2 build

v0.4.3

### Base game

RollerCoaster Tycoon 2, RollerCoaster Tycoon Classic

### Area(s) with this issue?

_No response_

### Describe the issue

If you're playing a park for a while and try to quit the game or load another park, the game nicely asks you if you don't want to save first. If you instead go directly for "new game", it just straight jumps into the new park without asking anything.

### Steps to reproduce

1. Open any park
2. Use hyper speed for a couple seconds to trigger the save fallback
3. Open the menu and request new game on a new park

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by nehemiah-negussie**
Opening new game would previously not prompt for anything. Add save prompt functionality similar to load game for new game. Added new PromptMode enum called SaveBeforeNew with appropriate string ids.

### CODE REVIEW

File: src/openrct2-ui/windows/SavePrompt.cpp [hunk 1]
```diff
@@ -64,6 +64,7 @@ static constexpr const StringId window_save_prompt_labels[][2] = {
     { STR_LOAD_GAME_PROMPT_TITLE,   STR_SAVE_BEFORE_LOADING },
     { STR_QUIT_GAME_PROMPT_TITLE,   STR_SAVE_BEFORE_QUITTING },
     { STR_QUIT_GAME_2_PROMPT_TITLE, STR_SAVE_BEFORE_QUITTING_2 },
+    { STR_NEW_GAME, STR_SAVE_BEFORE_QUITTING },
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/SavePrompt.cpp:67]:
```suggestion
    { STR_QUIT_GAME_2_PROMPT_TITLE, STR_SAVE_BEFORE_QUITTING_2 },
    { STR_NEW_GAME,                 STR_SAVE_BEFORE_QUITTING },
```

File: src/openrct2/Game.cpp [hunk 2]
```diff
@@ -736,6 +736,15 @@ static void game_load_or_quit_no_save_prompt_callback(int32_t result, const utf8
     }
 }
 
+static void game_new_game_window_callback(const utf8* path)
```
Comment (tupaschoal) [inline src/openrct2/Game.cpp:739]:
```suggestion
static void NewGameWindowCallback(const utf8* path)
```

We use TitleCase for functions (I know it is inconsistent here)

File: src/openrct2/interface/Window.h [hunk 3]
```diff
@@ -466,6 +466,7 @@ enum class PromptMode : uint8_t
     SaveBeforeLoad = 0,
     SaveBeforeQuit,
     SaveBeforeQuit2,
+    SaveBeforeNew,
```
Comment (tupaschoal) [inline src/openrct2/interface/Window.h:469]:
```suggestion
    SaveBeforeNewGame,
```

File: src/openrct2/Game.cpp [hunk 4]
```diff
@@ -778,6 +787,16 @@ void game_load_or_quit_no_save_prompt()
             title_load();
             break;
         }
+        case PromptMode::SaveBeforeNew:
```
Comment (tupaschoal) [inline src/openrct2/Game.cpp:790]:
```suggestion
        case PromptMode::SaveBeforeNewGame:
```

**Review by tupaschoal — COMMENTED** [review]
I've tested and it works smoothly, nice job!

My comments are just style consistency. Note you will have to add a changelog entry to `distribution/changelog.txt` too

**Comment by nehemiah-negussie** [comment]
Great I'll look into fixing these and commiting the changes. Thanks

File: distribution/changelog.txt [hunk 5]
```diff
@@ -11,6 +11,7 @@
 - Fix: [#18905] Ride Construction window theme is not applied correctly.
 - Fix: [#18911] Mini Golf station does not draw correctly from all angles.
 - Fix: [#19026] Park loan is clamped to a 32-bit integer.
+- Fix: [#18971] Fix New Game not prompting for save before quitting.
```
Comment (tupaschoal) [inline distribution/changelog.txt:14]:
```suggestion
- Fix: [#18971] New Game does not prompt for save before quitting.
```

Comment (tupaschoal) [inline distribution/changelog.txt:14]:
```suggestion
- Fix: [#18971] New Game does not prompt for save before quitting.
- Fix: [#19026] Park loan is clamped to a 32-bit integer.
```

**Review by tupaschoal — APPROVED** [review]

**Pull request merged**